pkg/engine/artifacts_test.go

```diff
@@ -0,0 +1,663 @@
+					KubernetesConfig: &api.KubernetesConfig{
+						EnablePodSecurityPolicy: to.BoolPtr(true),
+					},
+				},
+			},
+			expectedKubeDNS:            false,
+			expectedCoreDNS:            true,
+			expectedKubeProxy:          true,
+			expectedAzureNetworkPolicy: false,
+			expectedCilium:             false,
+			expectedFlannel:            false,
+			expectedAADAdminGroup:      false,
+			expectedAzureCloudProvider: true,
+			expectedAuditPolicy:        true,
+			expectedELBService:         false,
+			expectedPodSecurityPolicy:  true,
+		},
+	}
+
+	for _, c := range cases {
+		componentFileSpecArray := kubernetesAddonSettingsInit(c.p)
+		for _, componentFileSpec := range componentFileSpecArray {
+			switch componentFileSpec.destinationFile {
+			case "kube-dns-deployment.yaml":
+				if c.expectedKubeDNS != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", KubeDNSAddonName, c.expectedKubeDNS)
+				}
+			case "coredns.yaml":
+				if c.expectedCoreDNS != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", CoreDNSAddonName, c.expectedCoreDNS)
+				}
+			case "kube-proxy-daemonset.yaml":
+				if c.expectedKubeProxy != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", KubeProxyAddonName, c.expectedKubeProxy)
+				}
+			case "azure-npm-daemonset.yaml":
+				if c.expectedAzureNetworkPolicy != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", AzureNetworkPolicyAddonName, c.expectedAzureNetworkPolicy)
+				}
+			case "cilium-daemonset.yaml":
+				if c.expectedCilium != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", CiliumAddonName, c.expectedCilium)
+				}
+			case "flannel-daemonset.yaml":
+				if c.expectedFlannel != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", FlannelAddonName, c.expectedFlannel)
+				}
+			case "aad-default-admin-group-rbac.yaml":
+				if c.expectedAADAdminGroup != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", AADAdminGroupAddonName, c.expectedAADAdminGroup)
+				}
+			case "azure-cloud-provider-deployment.yaml":
+				if c.expectedAzureCloudProvider != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", AzureCloudProviderAddonName, c.expectedAzureCloudProvider)
+				}
+			case "audit-policy.yaml":
+				if c.expectedAuditPolicy != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", AuditPolicyAddonName, c.expectedAuditPolicy)
+				}
+			case "elb-svc.yaml":
+				if c.expectedELBService != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", ELBServiceAddonName, c.expectedELBService)
+				}
+			case "pod-security-policy.yaml":
+				if c.expectedPodSecurityPolicy != componentFileSpec.isEnabled {
+					t.Fatalf("Expected %s to be %t", "PodSecurityPolicy", c.expectedPodSecurityPolicy)
+				}
+			}
+		}
+	}
+}
+
+func TestKubernetesManifestSettingsInit(t *testing.T) {
+	// TODO add tests for kubernetesManifestSettingsInit
```

Comment:
The tests in here are 💯. Is this `// TODO:` for a followup PR?

Reply:
Sorry that wasn't clear. Yeah, because the "manifest" stuff is not strictly speaking addons (and this PR is large enough as-is) we'll track this in a future PR.